### PR TITLE
Add reloadMethod RPC

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

This method will allow the tool to replace hot reload with a more efficient implementation specifically for hot UI in the future. For now it delegates to reloadSources.

Note: removed unreachable type error checking

### CODE REVIEW

File: packages/flutter_tools/lib/src/vmservice.dart [hunk 1]
```diff
@@ -123,16 +123,9 @@ class VMService {
         final bool force = params.asMap['force'] ?? false;
         final bool pause = params.asMap['pause'] ?? false;
 
-        if (isolateId is! String || isolateId.isEmpty) {
+        if (isolateId.isEmpty) {
           throw rpc.RpcException.invalidParams('Invalid \'isolateId\': $isolateId');
         }
-        if (force is! bool) {
```
Comment (jacob314) [inline packages/flutter_tools/lib/src/vmservice.dart]: was this code intentionally removed? seems unrelated.

**Review by jacob314 — COMMENTED** [review]
a couple minor nits.

File: packages/flutter_tools/lib/src/vmservice.dart (hunk 1 above)
Comment (jacob314) [reply packages/flutter_tools/lib/src/vmservice.dart]: ah yes :) fine to remove.
We must not be running any lints on this code.

**Comment by jonahwilliams** [comment]
Updated to require class, library, method, and methodBody parameters
